### ISSUE TITLE
[Bug]: LocateProjects doesn't correctly handle project deletions.

### ISSUE THREAD

**Issue by IEvangelist** [dotnet/docs-tools#145]
## Discovery

I was investigating why four recent dependabot automated PRs were failing. Here is an example failed build from one of the: https://github.com/dotnet/docs/actions/runs/4355198732/jobs/7611491747.

I immediately found it odd that there were missing source code files. I discovered that these files and projects were deleted.

## Realization

The _LocateProjects_ app needs to account for project deletions.

Recently, there was a significant deletion PR in docs: https://github.com/dotnet/docs/pull/34146. This PR deleted several projects that were part of a solution. However, the snippets 5000 build validation didn't validate that the solution still compiled because the solution (*.sln) wasn't returned as part of the _LocateProjects_ subroutine.

In other words, _LocateProjects_ needs to be updated to return solutions when a project is deleted.

## Proposal

When a project is deleted, check to see if the said project was part of a solution. If it was, return the solution that it was a part of.

---
[Associated WorkItem - 85827](https://dev.azure.com/msft-skilling/Content/_workitems/edit/85827)

**Comment by adegeo** [comment]
Will handle as part of #113 

I've written a test case for the code base where this fails to pass. As I upgrade the tool I can validate that the test case passes and we'll know this scenario is covered.
